crates/krator-derive/src/admission.rs

```diff
@@ -0,0 +1,273 @@
+                    data.insert("tls.crt".into(), cert.serialize_pem().unwrap());
+                    data.insert("tls.key".into(), cert.serialize_private_key_pem());
+
+                    k8s_openapi::api::core::v1::Secret {
+                        metadata: k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta {
+                            name: Some(#name_identifier::admission_webhook_secret_name()),
+                            namespace: Some(namespace.to_string()),
+                            ..Default::default()
+                        },
+                        string_data: Some(data),
+                        type_: Some("tls".to_string()),
+                        ..Default::default()
+                    }
+                }
+
+                /// Returns a service that forwards to pods where label `app` has the value returned by the function
+                /// `admission_webhook_service_app_selector()`. It exposes port `443` (necessary for webhooks) and
+                /// listens to the pod's port `8443`
+                pub fn admission_webhook_service(namespace: &str) -> k8s_openapi::api::core::v1::Service {
+                    let crd = #name_identifier::crd();
+
+                    let mut selector = std::collections::BTreeMap::new();
+                    selector.insert("app".into(), #name_identifier::admission_webhook_service_app_selector());
+
+                    k8s_openapi::api::core::v1::Service {
+                        metadata: k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta {
+                            name: Some(#name_identifier::admission_webhook_service_name()),
+                            namespace: Some(namespace.to_string()),
+                            ..Default::default()
+                        },
+                        spec: Some(k8s_openapi::api::core::v1::ServiceSpec {
+                            selector: Some(selector),
+                            ports: Some(vec![k8s_openapi::api::core::v1::ServicePort{
+                                protocol: Some("TCP".to_string()),
+                                port: 443,
+                                target_port: Some(k8s_openapi::apimachinery::pkg::util::intstr::IntOrString::Int(8443)),
+                                ..Default::default()
+                            }]),
+                            type_: Some("ClusterIP".to_string()),
+                            ..Default::default()
+                        }),
+                        status: None
+                   }
+                }
+
+                /// Creates a MutatingWebhookConfiguration using the certificate from the given service and the service
+                /// of the given service as configuration
+                pub fn admission_webhook_configuration(service: k8s_openapi::api::core::v1::Service, secret: k8s_openapi::api::core::v1::Secret) -> anyhow::Result<k8s_openapi::api::admissionregistration::v1::MutatingWebhookConfiguration> {
```

Comment:
I think all of the error types should be `Box<dyn Error + Send + Sync>` for maximum capability. We do not want to force people deriving things to use `anyhow`. If we already are with krator, then this is fine for now

Reply:
🤯 I got to admit, that I don't understand this as I am kind of new to Rust ... if you give me a pointer, I am happy to adjust the code accordingly

Reply:
Hahah no problem. So when you are trying to implement errors in the most generic way possible (meaning that you aren't going to require the use of a specific library), you can change return type to `Result<k8s_openapi::api::admissionregistration::v1::MutatingWebhookConfiguration, Box<dyn Error + Send + Sync>>`. Boxing a type that implements a trait in rust is how dynamic dispatch works but it is also essentially the "generic error type" or lowest common denominator in Rust. We also require the `Send + Sync` trait bounds as all of these are being used in an async context and will likely need to be sent across `await`s.

Hopefully that makes more sense? Honestly, if you don't feel comfortable doing it, we can handle it in a follow up too.

Reply:
@thomastaylor312 thanks for taking the time explaining this. That is kind of what I thought this is about. My remaining problem was on how to create this kind of error.
It took me a while to find out but finally a colleague poked me to `Err("foo".into())`  ... this should be mentioned in every Rust-error handling tutorial / texts / docs ;) 

I did this in commit e8a64e2